.github/workflows/pytest_gpu.yml

```diff
@@ -0,0 +1,210 @@
+          HOROVOD_WITHOUT_MXNET: 1
+        run: |
+          python --version
+          pip --version
+          python -m pip install -U pip
+          cmake --version
+
+          echo "MARKERS:" $MARKERS
+
+          if [ "$PYTORCH" == "nightly" ]; then
+            cat requirements.txt | sed '/^torch[>=<]/d' | sed '/^torchtext[>=<]/d' | sed '/^torchvision[>=<]/d' | sed '/^torchaudio[>=<]/d' > requirements-temp && mv requirements-temp requirements.txt
+            pip install --pre torch torchtext torchvision torchaudio
+          else
+            pip install torch==$PYTORCH torchtext torchvision torchaudio
+          fi
+          pip install protobuf==3.20.1 # https://github.com/databrickslabs/dbx/issues/257
+
+          if [ "$RAY_VERSION" == "nightly" ]; then
+            # NOTE: hardcoded for python 3.9 on Linux
+            pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp39-cp39-manylinux2014_x86_64.whl
+          else
+            # installing `six` early resolves ModuleNotFound error in ray==2.1.0
+            pip install six
+            pip install ray==$RAY_VERSION
+          fi
+          ray_expected=$(python -c "import ray; print(ray.__version__)")
+
+          torch_expected=$(python -c "import torch; print(torch.__version__)")
+
+          pip install --no-build-isolation --no-use-pep517 ConfigSpace # temporary fix: https://github.com/automl/ConfigSpace/issues/173
+          pip install '.[test]'
+          pip list
+
+          python -c "import torch; assert torch.__version__ == \"$torch_expected\", f\"torch {torch.__version__} != $torch_expected\""
+          python -c "import ray; assert ray.__version__ == \"$ray_expected\", f\"ray {ray.__version__} != $ray_expected\""
+        shell: bash
+
+      - name: Install Neuropod backend
+        run: |
+          sudo mkdir -p "$NEUROPOD_BASE_DIR"
+          curl -L https://github.com/uber/neuropod/releases/download/v${{ env.NEUROPOD_VERSION }}/libneuropod-cpu-linux-v${{ env.NEUROPOD_VERSION }}-torchscript-${{ env.TORCHSCRIPT_VERSION }}-backend.tar.gz | sudo tar -xz -C "$NEUROPOD_BASE_DIR"
+        shell: bash
+
+      - name: Reinstall Horovod if necessary
+        env:
+          HOROVOD_WITH_PYTORCH: 1
+          HOROVOD_WITHOUT_MPI: 1
+          HOROVOD_WITHOUT_TENSORFLOW: 1
+          HOROVOD_WITHOUT_MXNET: 1
+        run: |
+          HOROVOD_BUILT=$(python -c "import horovod.torch; horovod.torch.nccl_built(); print('SUCCESS')" || true)
+          if [[ $HOROVOD_BUILT != "SUCCESS" ]]; then
+            pip uninstall -y horovod
+            pip install --no-cache-dir git+https://github.com/horovod/horovod.git@master
+          fi
+          horovodrun --check-build
+        shell: bash
+
+      - name: Check CUDA is available
+        run: |
+          python -c "import torch; assert torch.cuda.is_available()"
+
+      - name: Tests
+        run: |
+          pytest -v --timeout 300 --durations 10 -m "gpu" --junitxml pytest_gpu.xml tests
```

Comment:
Do you think we should mark specific tests with `gpu` or do we want to run the full suite?

Reply:
That's a great question– when you say the full suite, do you mean both `distributed` and `not distributed` tests? Or one of the two?

Reply:
Chatted with Travis about this– I think we'll do the full "not distributed" suite for this, since we don't necessarily want to spin up a multi-GPU machine for distributed GPU tests. Revised the PR. Thanks for bringing this up!